lib/new_relic/agent/instrumentation/grpc/server/instrumentation.rb

```diff
@@ -86,17 +97,23 @@ def set_host_and_port_and_method_info_on_desc
             end
           end
 
-          def server_options(headers)
+          def grpc_headers(metadata)
+            metadata.reject { |k, v| DT_KEYS.include?(k) }
+          end
+
+          def grpc_params(metadata, streamer_type)
             host = instance_variable_get(INSTANCE_VAR_HOST)
             port = instance_variable_get(INSTANCE_VAR_PORT)
             method = instance_variable_get(INSTANCE_VAR_METHOD)
-            {
-              request: {
-                headers: headers,
-                uri: "grpc://#{host}:#{port}/#{method}",
-                method: method
-              }
-            }
+            {'request.headers': grpc_headers(metadata),
+             'request.uri': "grpc://#{host}:#{port}/#{method}",
+             'request.method': method,
+             'request.grpc_type': streamer_type}
+          end
+
+          def trace_options
+            method = instance_variable_get(INSTANCE_VAR_METHOD)
+            {category: CATEGORY, transaction_name: "Controller/#{method}"}
```

Comment:
Could you add your notes about why "Controller/" has to prefix the transaction name?

Reply:
> Could you add your notes about why "Controller/" has to prefix the transaction name?

`Transaction#summary_metric`s requires `@frozen_name` to start with `Controller` in order to make use of the `HttpDispatcher` metric. If the name starts with anything else, the summary metrics will be background job related.